aten/src/ATen/native/cuda/TensorCompare.cu

```diff
@@ -40,4 +215,148 @@ Tensor _s_where_cuda(
+  return cmp_cuda<std::greater>(self, other, "gt");
+}
+
+Tensor le_cuda(const Tensor& self, const Tensor& other) {
+  return cmp_cuda<std::less_equal>(self, other, "le");
+}
+
+Tensor ge_cuda(const Tensor& self, const Tensor& other) {
+  return cmp_cuda<std::greater_equal>(self, other, "le");
+}
+
+Tensor eq_cuda(const Tensor& self, const Tensor& other) {
+  return cmp_cuda<std::equal_to>(self, other, "eq");
+}
+
+Tensor ne_cuda(const Tensor& self, const Tensor& other) {
+  return cmp_cuda<std::not_equal_to>(self, other, "ne");
+}
+
+Tensor& lt_out_cuda(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cuda<std::less>(result, self, other, "lt");
+}
+
+Tensor& gt_out_cuda(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cuda<std::greater>(result, self, other, "gt");
+}
+
+Tensor& le_out_cuda(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cuda<std::less_equal>(result, self, other, "le");
+}
+
+Tensor& ge_out_cuda(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cuda<std::greater_equal>(result, self, other, "ge");
+}
+
+Tensor& eq_out_cuda(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cuda<std::equal_to>(result, self, other, "eq");
+}
+
+Tensor& ne_out_cuda(Tensor& result, const Tensor& self, const Tensor& other) {
+  return cmp_out_cuda<std::not_equal_to>(result, self, other, "ne");
+}
+
+Tensor& lt_inplace_cuda(Tensor& self, Scalar other) {
+  return cmp_inplace_cuda<std::less>(self, other, "lt_");
+}
+
+Tensor& gt_inplace_cuda(Tensor& self, Scalar other) {
+  return cmp_inplace_cuda<std::greater>(self, other, "gt_");
+}
+
+Tensor& le_inplace_cuda(Tensor& self, Scalar other) {
+  return cmp_inplace_cuda<std::less_equal>(self, other, "le_");
+}
+
+Tensor& ge_inplace_cuda(Tensor& self, Scalar other) {
+  return cmp_inplace_cuda<std::greater_equal>(self, other, "ge_");
+}
+
+Tensor& eq_inplace_cuda(Tensor& self, Scalar other) {
+  return cmp_inplace_cuda<std::equal_to>(self, other, "eq_");
+}
+
+Tensor& ne_inplace_cuda(Tensor& self, Scalar other) {
+  return cmp_inplace_cuda<std::not_equal_to>(self, other, "ne_");
+}
+
+Tensor& lt_inplace_cuda(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cuda<std::less>(self, other, "lt_");
+}
+
+Tensor& gt_inplace_cuda(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cuda<std::greater>(self, other, "gt_");
+}
+
+Tensor& le_inplace_cuda(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cuda<std::less_equal>(self, other, "le_");
+}
+
+Tensor& ge_inplace_cuda(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cuda<std::greater_equal>(self, other, "ge_");
+}
+
+Tensor& eq_inplace_cuda(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cuda<std::equal_to>(self, other, "eq_");
+}
+
+Tensor& ne_inplace_cuda(Tensor& self, const Tensor& other) {
+  return cmp_inplace_cuda<std::not_equal_to>(self, other, "ne_");
+}
```

Comment:
This file is almost the same as the CPU one, so it would be good to avoid duplication (it's very easy to make the special cases for op templates inconsistent between devices). Do we have any ways to do this?